editor/src/clj/editor/settings_core.clj

```diff
@@ -38,22 +55,26 @@
   (BufferedReader. (StringReader. content)))
 
 (defn- empty-parse-state []
-  {:current-category nil :settings nil})
+  {:current-category nil :settings []})
 
 (defn- parse-category-line [{:keys [current-category settings] :as parse-state} line]
   (when-let [[_ new-category] (re-find #"^\s*\[([^\]]*)\]" line)]
     (assoc parse-state :current-category new-category)))
 
-(defn- parse-setting-line [{:keys [current-category settings] :as parse-state} line]
-  (when-let [[_ key val] (seq (map s/trim (re-find #"([^=]+)=(.*)" line)))]
-    (when-let [setting-path (seq (non-blank (s/split key #"\.")))]
-      (update parse-state :settings conj {:path (into [current-category] setting-path) :value val}))))
+ (defn- parse-setting-line [{:keys [current-category settings] :as parse-state} line]
+   (when-let [[_ key val] (seq (map s/trim (re-find #"([^=]+)=(.*)" line)))]
+     (when-let [setting-path (seq (non-blank (s/split key #"\.")))]
+       (let [settings (get-in parse-state [:settings])
+             full-path (seq (conj [current-category] (first setting-path)))]
+         (if-let [existing-value (get-setting settings full-path)]
+           (update parse-state :settings set-setting full-path (str existing-value "," val))
+           (update parse-state :settings conj {:path full-path :value val}))))))
```

Comment:
We had an offline discussion that this function will probably need to be rewritten somewhat to support game property paths with dots in them. Some feedback and thoughts:

* You already have `settings` from the function arguments, so no need to `(get-in...)`. in cases where the `get-in` is just one token, you can use `(get coll token)` or even `(coll token)` and avoid creating some garbage. If the token is a `:keyword`, we often choose to reverse the order as `(:keyword coll)`.
* There are a few places where you wrap an expression that returns a collection in `(seq ...)`. The reason it is done for the `setting-path` above is that `seq` will return `nil` if the collection is empty. In that case we want the `nil` because it evaluates to false, skipping the `when-let`. But the call to `(seq ...)` can be removed otherwise.
* In fact, the `full-path` can be expressed as just `[current-category (first setting-path)]` in the above code. However, we will likely need to keep more than just the first element from `setting-path` to support `.`-separated game property paths.
* The way I see it, we should do the split on `.` like we used to, and after that check if `(last setting-path)` ends in `#<number>`. You can use `re-find` and destructuring to achieve this. Something along the lines of:
```clj
(let [setting-path (vec setting-path)] ; Easier to work with a vector.
  (if-let [[_ clean-path-token] (re-find #"(.*)#\d+$" (peek setting-path))]
    (let [clean-path (conj (pop setting-path) clean-path-token)]
      (append-to-comma-separated-string-at clean-path ...))
    (as-before setting-path ...)))
```

Reply:
Thanks! Really helpful explanations!

I changed to # as the separator. And I did a somewhat different approach from yours. I start by splitting the string on # and getting the `first` part. The rest is basically the same as before where I split on . and use all parts to construct the path.